src/features_finder.rs

```diff
@@ -87,6 +87,21 @@ pub fn fetch_feature_sets(package: &crate::cargo_metadata::Package) -> Vec<Featu
                 // skip_feature_set matches: do not add it to feature_sets
                 continue 'outer;
             }
+            for feat in feature_set.clone() {
+                for dep in package
+                    .feature_map
+                    .get(&feat.0)
+                    .unwrap_or(&FeatureList::default())
+                    .iter()
+                {
+                    if dep.strip_prefix("dep:").is_some() {
+                        continue;
+                    }
+                    if !feature_set.contains(&dep) {
+                        continue 'outer;
```

Comment:
thought: it would be nice if these skipped feature combos were still included in the output at the bottom, with an explanation of which run they are equivalent to. unsure if that's going to be really annoying to do properly, could be fine with "skipped covered feature set [...] due to dependency from .. to .."

Reply:
Looking at my follow-up work regarding the rule abstraction, those dependencies would be encoded with the same rule mechanism as other selection rules, e.g. skip_feature_sets, denylists etc. 
For this reason, I would not want to build this into this for-loop, but maybe externally creating a list of skipped feature sets because of dependencies. This could even be generalized, to print for every skipped feature set one or even all rules, that result in the set skipped. But I also think that in normal use cases, this output is unwanted.

What you are probably thinking of is helping users understand all the rules applied to the feature set selection and help "debugging" those rules. Probably, this workflow is started by recognizing a specific feature set is not run while the dev does not understand why it is skipped. So the dev wants to "ask" why a certain feature set was skipped. We could provide a feature flag, taking an feature set as argument value, and we print all the conflicting rules to the screen, if it is skipped.

I think this would be an easy and nice addition to the tool, but I would like to implement it on top of the rule abstraction, because as of now, this would be quite tedious handling every "rule" case separatly.

Reply:
Yes, that's fine, also this was a "thought"; I wasn't requiring this of this PR.

I'm also not sure we should print _every_ skipped set.